### PR TITLE
feature/insertion-test-cases

### LINKED ISSUE #1: Write Unit tests to cover basic functionality
Aim is to cover basic functionality with unit tests using [bats](https://github.com/sstephenson/bats)

Basic outline of what we have to cover:

• insertion into target
• insertion into group ( 1. # comment group, 2. def group )
• insertion into root
• reserved word skipping tests (repeat insertion test but with condition to skip reserved words)
• dependency options insertion, check that all pod options are inserted correctly, for example:
```bash
pod 'Alamofire', path => '/some/local/path' # ensure that path option is correct
```
(You can follow [Podfile syntax reference](https://guides.cocoapods.org/syntax/podfile.html#pod) to ensure correctness of options being tested)

### LINKED ISSUE #6: Add installation via homebrew
Homebrew is mostly preferred (and easiest) way of installing command-line tools.
But adding it to default homebrew formulae `homebrew/homebrew-core` requires project to be somehow popular and used.

Though we have an alternative: we can create our own formulae trunk (called `tap`) and our installation formulae there.

https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://docs.brew.sh/Formula-Cookbook

### PR DESCRIPTION
**Pull request by rymbekov**
closes #1 
closes #6

### CODE REVIEW

File: test/insertion.bats [hunk 1]
```diff
@@ -1,38 +1,111 @@
 load test_helper
 
-@test "insertion target" {
-    result="$(p -a Alamofire -t Altel -f $suts/podfile-altel -d)"
+setup() {
```
Comment (aidarnouman) [inline test/insertion.bats]: I think it'd better if we move this to `test_helper.sh`

File: test/test_helper.bash [hunk 2]
```diff
@@ -5,3 +5,20 @@ suts=$BATS_TEST_DIRNAME/suts
...
+
+function init {
+    touch $suts/temp
+    echo "use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+
+target 'Altel' do
+  pod 'R.swift'
+  pod 'SwiftFormat/CLI'
```
Comment (aidarnouman) [inline test/test_helper.bash]: I think we can remove this pod right here, it doesn't help us in any way

File: test/test_helper.bash [hunk 3]
```diff
@@ -5,3 +5,20 @@ suts=$BATS_TEST_DIRNAME/suts
...
     bash "$(dirname $BATS_TEST_DIRNAME)"/libexec/p "$@"
 }
+
+function init {
+    touch $suts/temp
+    echo "use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+
+target 'Altel' do
```
Comment (aidarnouman) [inline test/test_helper.bash]: Also please rename the target name :) 
(NDA stuff you know)

File: test/insertion.bats [hunk 4]
```diff
@@ -1,38 +1,111 @@
...
+end
+
+def insert_pods
+  pod 'Typhoon', '~>4.0’
+  # Analytics
+end"
+[ "$result" == "$text" ]
+}
+
+@test "skipping reserved word" {
```
Comment (aidarnouman) [inline test/insertion.bats]: I think we can move these testcases to separate file and split them one by one.

File: test/suts/podfile-default [hunk 5]
```diff
@@ -0,0 +1,12 @@
+use_frameworks!
```
Comment (aidarnouman) [inline test/suts/podfile-default]: This is file is no longer needed, I think

File: test/insertion.bats (hunk 4 above)
Comment (aidarnouman) [reply test/insertion.bats]: Also after moving to new file we need to add `test` command inside `Makefile` which runs all tests inside `test/` folder.

File: test/insertion.bats [hunk 6]
```diff
@@ -1,38 +1,111 @@
...
+  result="$(p -a Alamofire -t Altel -f $suts/temp)"
+  result="$(p -a Alamofire -g insert_pods -f $suts/temp -d)"
+
+  text="use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+pod 'Alamofire'
+
+target 'Altel' do
+pod 'Alamofire'
```
Comment (aidarnouman) [inline test/insertion.bats]: You need to modify indentation logic so that it would work correctly - and fix tests appropriately.

**Review by nugmanoff — CHANGES_REQUESTED** [review]

**Comment by nugmanoff** [comment]
Please remove all the test-podfiles from `suts/` directory. Also you can create `temp` file directly in the `test/` folder, because `suts/` directory will no longer be needed.

**Comment by nugmanoff** [comment]
You need to rebase from master and extend `Makefile` to include `test` command as I described above. It is pretty straightforward.

**Comment by nugmanoff** [comment]
One more suit of testcases is needed, I mentioned it in #1 

> • dependency options insertion, check that all pod options are inserted correctly, for example:
> `pod 'Alamofire', path => '/some/local/path' # ensure that path option is correct`

You can create it in separate file as well in order to keep things clean and do not bloat one particular file.

**Comment by nugmanoff** [comment]
Please rename the PR name appropriately according to branch name, you made a mistake in delimiters there :)

**Comment by nugmanoff** [comment]
All in all, you both did a great job on this one @nurbekismagulov @rymbekov !
Thanks a lot, brothers! 🔥 
 Keep it going 💪 🚀

File: test/test_helper.bash (hunk 2 above)
Comment (nurbekismagulov) [reply test/test_helper.bash]: I left this pod to make sure that slash symbol doesn’t interfere with the function. I'll remove it

File: test/insertion.bats (hunk 4 above)
Comment (nurbekismagulov) [reply test/insertion.bats]: We think that separating tests into different files is not good idea. Isn't it enough to have one file and run it inside Makefile?

Comment (aidarnouman) [reply test/insertion.bats]: Separation is generally a good idea when it comes to file organization. No one loves bloated stuff :)
Usually you can write most of the testcases in one file, but in most of the testing frameworks there is a way to group different testing scenarios for better readability of the test code as well as better readability of the output.

Here is the example usage of Jest (JS testing framework):

```javascript
describe('testing arithmetical functions', () => {
  it('should add numbers properly', () => {
     expect(2+2).toBeEqual(4)
   });
   it('should multiply numbers properly', () => {
     expect(3*2).toBeEqual(6)
   });
});

describe('testing some other stuff', () => {
 // ...
});

```

Here we can clearly separate two different scenarios "testing arithmetic functions" and "some other stuff". Output of above testcases would also be properly formatted in the console. As well as it provides better readability for the developers. But due to the fact that we are limited with what `bats` provides us here - we need to opt for just moving grouped testcases to separate files :)

File: test/test_helper.bash (hunk 2 above)
Comment (aidarnouman) [reply test/test_helper.bash]: @nurbekismagulov btw, very nice point! I didn't even think about it. Keep it then, if you feel like so

**Review by nugmanoff — APPROVED** [review]

**Pull request merged**